tests/test_health_gating.matrix

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env matrix
+"tests":
+- "name": Verify gating
+  "description": >
+    This test should fail! We're verifying that our gating on health works.
```

Comment:
If core ever fixes the issue with trusty, this will stop failing.  At some point it would probably be worth adding a `force_unhealthy` task that uses `juju_model.run('status-set error')` (if that works).  Then we could use the basic bundle as well to make the test (marginally) faster.